ui/components/component-library/avatar-account/avatar-account.tsx

```diff
@@ -6,13 +6,30 @@ import BlockieIdenticon from '../../ui/identicon/blockieIdenticon/blockieIdentic
 import type { PolymorphicRef } from '../box';
 
 import { AvatarBase, AvatarBaseProps } from '../avatar-base';
+import {
+  getCaipNamespaceFromAddress,
+} from '../../../../shared/lib/multichain';
 import {
   AvatarAccountDiameter,
   AvatarAccountVariant,
   AvatarAccountSize,
   AvatarAccountComponent,
   AvatarAccountProps,
 } from './avatar-account.types';
+import { KnownCaipNamespace } from '@metamask/utils';
+
+function getJazziconNamespace(address: string): string | undefined {
+  const namespace = getCaipNamespaceFromAddress(address);
+
+  switch (namespace) {
+    case KnownCaipNamespace.Bip122:
+      return namespace;
+    case KnownCaipNamespace.Eip155:
+      return undefined; // Falls back to default Jazzicon behavior
+    default:
+      return undefined;
+  }
+}
```

Comment:
Question: Why is `getJazziconNamespace` defined in the `AvatarAccount` component instead of within the `Jazzicon` component itself? 

This function appears to be specific to Jazzicon behavior and isn't used for Blockies. It seems like it might be more appropriate to encapsulate this logic within the `Jazzicon` component for better separation of concerns and to keep the `AvatarAccount` component more generic?

Reply:
Actually, you are 100% right, this makes more sense to move it there. I'm changing that!